guides/plugins/apps/administration/add-custom-modules.md

```diff
@@ -0,0 +1,150 @@
+
+```javascript
+function sendReadyState() {
+    window.parent.postMessage('sw-app-loaded', '*');
+}
+```
+
+This has to be done as soon as everything is loaded so that the loading spinner disappears. If your view is not fully loaded after 5 seconds, it will be aborted.
+
+## Structure your modules
+
+With Shopware 6.4.0.0 we added a third level in the admin menu structure. This change was made to give you as a developer the opportunity to group your administration modules if needed.
+
+When you define a module it gets automatically loaded by the administration. Additionally the administration creates a menu entry for your module. You can reference this menu entry and set it as the parent menu entry for your other modules.
+
+The navigation id of your modules is always in the form `app-<appName>-<moduleName>`. So, within your manifest you can add a reference to modules that you just created:
+
+{% code title="manifest.xml" %}
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<manifest xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/master/src/Core/Framework/App/Manifest/Schema/manifest-1.0.xsd">
+    <meta>
+        <name>myApp</app>
+        ...
+    </meta>
+    <admin>
+        <module name="myModules"
+                source="https://example.com/promotion/view/promotion-module"
+                parent="sw-catalogue"
+                position="50"
+        >
+            <label>My apps modules</label>
+            <label lang="de-DE">Module meiner app</label>
+        </module>
+
+        <module name="someModule"
+                source="https://example.com/promotion/view/promotion-module"
+                parent="app-myApp-myModules"
+                position="1"
+        >
+            <label>Module underneath "My apps modules"</label>
+            <label lang="de-DE">Modul unterhalb von "Module meiner app"</label>
+        </module>
+    </admin>
+</manifest>
+
+```
+{% endcode %}
+
+Modules that are used as a parent for other modules do not need the `source` attribute to be set, although they can.
+
+## Add main module to your app
+
+With Shopware 6.4.0. You can define a main module for your app. This "special" module will be opened from the list of your installed apps as well as from the app detail page if you bougth it from the Shopware store.
+
+Your main module can be defined by adding a `main-module` element within your `administration` section of your manifest file. It's only required attribute is the `source` attribute.
+
+To avoid mixing other modules with your main module we decided to separate the main module from modules with navigation entries. You can still use the same URL on both, a module that is available through the menu and your main module.
```

Comment:
```suggestion
To avoid mixing other modules with your main module, we decided to separate the main module from modules with navigation entries. You can still use the same URL on both, a module that is available through the menu and your main module.
```